kafka_utils/kafka_rolling_restart/precheck.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import shlex
+
+
+class PrecheckFailedException(Exception):
+    pass
+
+
+class Precheck(object):
```

Comment:
What about call this Task and we can then distinguish between LocalPreStopTask and RemotePreStopTask. LocalPreTask is executed locally while remote is executed on the remote broker. I'd suggest to also add RemotePostStopTask and LocalPostStopTask (executed after the stop but before the start of the Kafka broker).

This would allow enable users to do tasks such as instance type bump, instance reboot (ex. to install a new driver), etc.